extensions/civetweb/processors/ListenHTTP.cpp

```diff
@@ -191,6 +211,16 @@ void ListenHTTP::onSchedule(core::ProcessContext *context, core::ProcessSessionF
   server_.reset(new CivetServer(options));
   handler_.reset(new Handler(basePath, context, sessionFactory, std::move(authDNPattern), std::move(headersAsAttributesPattern)));
   server_->addHandler(basePath, handler_.get());
+
+  if (randomPort) {
+    const auto& vec = server_->getListeningPorts();
+    if (vec.size() != 1) {
+      logger_->log_error("Random port is set, but there is no listening port! Server most probably failed to start!");
```

Comment:
If this is listening on http and https and a test will use either/both how will it know those random ports?

Reply:
There is no such option. 
The processor currently only supports one port property, which is validated using port validator.
http XOR https depends on whether SSL cert property is provided or not. In case it is, the processor automatically switches to https. 
To make sure I will add a getter for this, too.